### PR TITLE
 Build & test update 

### PR DESCRIPTION
**Pull request by odinuv**
Preparation for
https://keboola.atlassian.net/browse/AIS-35

- updated code climate
- added .env & .env setup to easily run tests locally
- fixed dockerhub setup
- made SKIP_ABS_TESTS required for clarity
- updated python version
- removed obsolete stuff
- fix handling of root URL with trailing slash
- fixed test_custom_headers test due to changes in connection
- moved test_load_files_to_workspace to a separate file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,101 +1,4 @@
...
 __pycache__/
-*.py[cod]
-*$py.class
-
-# C extensions
-*.so
-
-# Distribution / packaging
-.Python
-build/
```
Comment (pivnicek) [inline .gitignore:12]: I think we want to keep skipping the build dir?  At least I was using it before I think

File: .env.template [hunk 2]
```diff
@@ -0,0 +1,4 @@
+KBC_TEST_TOKEN=
+# URL without trailing slash
```
Comment (pivnicek) [inline .env.template:2]: imo comment not necessary anymore because your rtrimming the trailing slash now below in the client constructor

**Review by pivnicek — APPROVED** [review]
Tidy bit of business 👍

File: .gitignore (hunk 1 above)
Comment (odinuv) [reply .gitignore:12]: Looks like it's not being made anymore

File: .env.template (hunk 2 above)
Comment (odinuv) [reply .env.template:2]: Yeah, well the client works without the slash, but the tests don't, that would need to make some assertions more benevolent - so yes, it is necessary at the moment

**Pull request merged**